crates/net/network/src/transactions/mod.rs

```diff
@@ -1102,21 +1108,44 @@ where
                     metrics_increment_egress_peer_channel_full,
                 );
             }
-            // drain commands
+
+            // advance commands (propagate/fetch/serve txns)
             if let Poll::Ready(Some(cmd)) = this.command_rx.poll_next_unpin(cx) {
                 this.on_command(cmd);
                 some_ready = true;
             }
 
-            // drain incoming transaction events
+            // advance incoming transaction events (stream new txns/announcements from network
+            // manager and queue for import to pool/fetch txns)
+            //
+            // can potentially queue around 1,5 k txns for import to pool if txns are valid. each
+            // `Transactions` message could contain at least up to soft limit byte size for
+            // broadcast message / avg small legacy tx byte size: 128 KiB / 100 bytes < 1,5k
+            // transactions.
+            //
+            // if txns however are invalid, and just 1 byte, since this isn't validated until
+            // import to pool, this can potentially queue around > 130 k txns. more if the
+            // message size is bigger than 128 KiB.
+            //
+            // this will potentially remove hashes from hashes pending fetch, it the event is an
+            // announcement (if same hashes are announced that didn't fit into a previous request)
```

Comment:
coming back to this, my main issue with this comment is that this doesn't make it super clear right away that these calculations are based on the soft limit and assumptions on tx size.

This section should be structured differently, starting from the soft limit so that it's immediately clear that the limiting factor is the size of the message.
Then we can keep the calculation as an example.

I'm also fine with keeping a note that this can be potentially larger depending on the actual size of the transaction, limited by the smallest footprint.

But I'd like to remove the "1 byte" and "130k" part, because mentioning these doesn't add real meaningful context and can be a bit confusing.

Reply:
I think exact values is exactly what we need to surpass reth from some devp2p impl to an exceptional client. I know it can be tedious and sometimes maybe intimidating to actually poke the code and see what it actually means, i.e. what the concrete axioms are, but this is the only way to really paradigm shit performance (pun on paradigm not intended, but it's cute, I like it).

instead of 1 byte, I will use 10 bytes which is correctly the minimum length of a txn that will decode into the first reth transactions type `TrascationSigned` that chains down to the type that is processed here `Transactions`. https://github.com/paradigmxyz/reth/pull/6853